koku/masu/processor/gcp/gcp_report_processor.py

```diff
@@ -0,0 +1,184 @@
+            self.column_map = report_common_db.column_map
+
+        LOG.info('Initialized report processor for file: %s and schema: %s',
+                 report_path, self._schema_name)
+
+    def _get_or_create_cost_entry_bill(self, row):
+        """Get or Create a GCP cost entry bill object.
+
+        Args:
+            row (OrderedDict): A dictionary representation of a CSV file row.
+
+        Returns:
+             (reporting.provider.gcp.models.GCPCostEntryBill) A GCP Bill instance.
+
+        """
+        data = self._get_data_for_table(row, GCPCostEntryLineItemDaily._meta.db_table)
+
+        report_date_range = utils.month_date_range(parser.parse(data['start_time']))
+        start_date, end_date = report_date_range.split('-')
+
+        start_date_utc = parser.parse(start_date).replace(hour=0, minute=0, tzinfo=pytz.UTC)
+        end_date_utc = parser.parse(end_date).replace(hour=0, minute=0, tzinfo=pytz.UTC)
+
+        data['billing_period_start'] = datetime.strftime(start_date_utc, '%Y-%m-%d %H:%M%z')
+        data['billing_period_end'] = datetime.strftime(end_date_utc, '%Y-%m-%d %H:%M%z')
+
+        bill, created = GCPCostEntryBill.objects.get_or_create(
+            billing_period_start=data['billing_period_start'],
+            provider=self.provider,
+            billing_period_end=data['billing_period_end']
+        )
+        return bill
+
+    def _get_or_create_gcp_project(self, row):
+        """Get or Create a GCPProject.
+
+        Args:
+            row (OrderedDict): A dictionary representation of a CSV file row.
+
+        Returns:
+             (reporting.provider.gcp.models.GCPProject) A GCP Project instance with project_id matching row_id.
+
+        """
+        data = self._get_data_for_table(row, GCPProject._meta.db_table)
+
+        project, created = GCPProject.objects.get_or_create(
+            account_id=data['account_id'],
+            project_id=data['project_id'],
+            defaults={
+                'project_name': data['project_name'],
+                'project_number': data['project_number'],
+                'project_labels': data['project_labels']
+            }
+        )
+        return project
+
+    def _create_cost_entry_line_item(self, row, bill, project):
+        """Create a cost entry line item object.
+
+        Args:
+            row (OrderedDict): A dictionary representation of a CSV file row
+            bill (reporting.provider.gcp.models.GCPCostEntryBill): A monthly GCPCostEntryBill
+            project (reporting.provider.gcp.models.GCPProject): A GCP Project
+
+        """
+        data = self._get_data_for_table(row, GCPCostEntryLineItemDaily._meta.db_table)
+        if not GCPCostEntryLineItemDaily.objects.filter(
+                project=project,
+                line_item_type=data['line_item_type'],
+                start_time=data['start_time']
+        ).exists():
+            line_item = GCPCostEntryLineItemDaily(
```

Comment:
Processing line items using the ORM is going to kill us on performance. Right now for AWS and Azure we are using a bulk csv read method which is about as fast as you can load bulk data into the DB. See https://github.com/project-koku/koku/blob/master/koku/masu/database/report_db_accessor_base.py#L166

We are already hitting some processing time concerns with AWS. I did some testing with this in the AWS processor early this week with a large file, trying the ORM just for a benchmark. 

With the bulk upload I was able to process the file in `224 seconds`
With the ORM and the exact same bulk create setup you have here I went up to `729 seconds`. 

So I think we really, really need to stick with the csv upload method.

Reply:
@adberglund AWS has monthly data right? For GCP data comes in daily csvs.
Since each batch of data to process is going to be a lot smaller would this still be a concern for GCP?

Reply:
@werwty Really depends on the account and the data source. So far in general we have under-estimated account sizes and that is currently biting us. 

It also looks like we might need to do the BigQuery export to get true line item data and all of the columns we will likely need. That will bump up the file size, too.

Reply:
I suppose that's fair enough. I'll rewrite this to use `bulk_insert_rows`